src/overlays/actors/ovl_Bg_Ikninside/z_bg_ikninside.c

```diff
@@ -28,30 +32,125 @@ const ActorInit Bg_Ikninside_InitVars = {
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80C076A8 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_PLAYER, OC2_TYPE_1, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x80000008, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+static Gfx* D_80C076A0[] = { object_ikninside_obj_DL_00A748, object_ikninside_obj_DL_00A5A8 };
+
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_PLAYER,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x80000008, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 32, 32, 0, { 0, 0, 0 } },
 };
 
-#endif
-
-extern ColliderCylinderInit D_80C076A8;
-
-extern UNK_TYPE D_0600CC78;
-extern UNK_TYPE D_0600DE48;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikninside/BgIkninside_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikninside/BgIkninside_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikninside/func_80C07220.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikninside/func_80C07230.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikninside/func_80C072D0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikninside/BgIkninside_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikninside/BgIkninside_Draw.s")
+static Vec3f D_80C076D4 = { 0.0f, -1.0f, 0.0f };
+
+void BgIkninside_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkninside* this = THIS;
+    CollisionHeader* colHeader = NULL;
+    s16 pad;
+
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    this->actionFunc = func_80C072D0;
+    DynaPolyActor_Init(&this->dyna, 0);
+    CollisionHeader_GetVirtual(&object_ikninside_obj_Colheader_00DE48, &colHeader);
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+    if (Flags_GetSwitch(globalCtx, DMIKNINSIDE_GET_SWITCH(thisx))) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    }
+}
+
+void BgIkninside_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkninside* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80C07220(BgIkninside* this, GlobalContext* globalCtx) {
+}
+
+void func_80C07230(BgIkninside* this, GlobalContext* globalCtx) {
+    if (this->dyna.actor.cutscene == -1) {
+        this->actionFunc = func_80C07220;
+    } else if (ActorCutscene_GetCurrentIndex() == 0x7c) {
+        ActorCutscene_Stop(0x7c);
```

Comment:
```suggestion
    } else if (ActorCutscene_GetCurrentIndex() == 0x7C) {
        ActorCutscene_Stop(0x7C);
```
Our style has Hex as uppercase letters.